setup.py

```diff
@@ -4,34 +4,148 @@
+else:
+
+    class UniversalBdistWheel(bdist_wheel):
+        def get_tag(self):
+            return (
+                'py2.py3',
+                'none',
+            ) + bdist_wheel.get_tag(self)[2:]
+
+    cmdclass['bdist_wheel'] = UniversalBdistWheel
+
+
+class SharedLibrary(Extension):
+    """Object that describes the library (filename) and how to make it.
+
+    Note: The `output_dir` parameter shouldn't include the final lib/ or bin/
+    component, we figure that out on the fly.
+    """
+    if sys.platform == "darwin":
+        prefix = "lib"
+        suffix = ".dylib"
+        lib_dir = "bin"
+    elif sys.platform == "win32":
+        prefix = ""  # VC prefers to leave out the "lib" prefix.
+        suffix = ".dll"
+        lib_dir = "bin"  # VC likes to put DLLs into PREFIX/bin instead.
+    else:
+        prefix = "lib"
+        suffix = ".so"
+        lib_dir = "lib"
+
+    def __init__(self, name, cmd, cwd=".", output_dir=".", env=None):
+        Extension.__init__(self, name, sources=[])
+        self.cmd = cmd
+        self.cwd = path.normpath(cwd)
+        self.output_dir = path.join(path.normpath(output_dir), self.lib_dir)
+        self.env = env or dict(os.environ)
+
+
+class SharedLibBuildExt(build_ext):
+    """Object representing command to produce and install a shared library."""
+
+    # Apparently needed for determining the correct library filename on
+    # Windows.
+    def get_ext_filename(self, ext_name):
+        for ext in self.extensions:
+            if isinstance(ext, SharedLibrary):
+                return (ext.prefix + os.path.join(*ext_name.split('.')) +
+                        ext.suffix)
+        return build_ext.get_ext_filename(self, ext_name)
+
+    def build_extension(self, ext):
+        if not isinstance(ext, SharedLibrary):
+            build_ext.build_extension(self, ext)
+            return
+
+        distutils.log.info("running '{}'".format(ext.cmd))
+        if not self.dry_run:
+            rv = subprocess.Popen(
+                ext.cmd, cwd=ext.cwd, env=ext.env, shell=True).wait()
+            if rv != 0:
+                sys.exit(rv)
+
+        lib_name = ext.prefix + ext.name.split(".")[-1] + ext.suffix
+        lib_fullpath = path.join(ext.output_dir, lib_name)
+        dest_path = self.get_ext_fullpath(ext.name)
+
+        distutils.dir_util.mkpath(
+            path.dirname(dest_path),
+            verbose=self.verbose,
+            dry_run=self.dry_run)
+
+        distutils.file_util.copy_file(
+            lib_fullpath,
+            dest_path,
+            verbose=self.verbose,
+            dry_run=self.dry_run)
+
+
+cmdclass['build_ext'] = SharedLibBuildExt
+
+if os.environ.get("FREETYPEPY_BUNDLE_FT"):
+    print("# Will build and bundle FreeType.")
+    ext_modules = [
+        SharedLibrary(
+            "freetype.freetype",  # double name and dot are needed for macOS.
+            cmd="python ./setup-build-freetype.py",
```

Comment:
why not call the script just "build_freetype.py"?

Reply:
also, you should use sys.executable, not "python", so it will execute the script with the same python interpreter used to run the setup.py

Reply:
I did that at some point and tox tripped over somehow.. but gonna try again.

I suppose I can shorten the name. Just looks nicer next to setup.py ;)

Reply:
> tox tripped over somehow

try passing to subprocess a list of strings, instead of a single string